chain/solana/src/chain.rs

```diff
@@ -0,0 +1,299 @@
+        }
+    }
+}
+
+#[async_trait]
+impl Blockchain for Chain {
+    const KIND: BlockchainKind = BlockchainKind::Solana;
+    type Block = codec::Block;
+    type DataSource = DataSource;
+    type UnresolvedDataSource = UnresolvedDataSource;
+    type DataSourceTemplate = DataSourceTemplate;
+    type UnresolvedDataSourceTemplate = UnresolvedDataSourceTemplate;
+    type TriggersAdapter = TriggersAdapter;
+    type TriggerData = crate::trigger::SolanaTrigger;
+    type MappingTrigger = crate::trigger::SolanaTrigger;
+    type TriggerFilter = crate::adapter::TriggerFilter;
+    type NodeCapabilities = crate::capabilities::NodeCapabilities;
+    type RuntimeAdapter = RuntimeAdapter;
+
+    fn triggers_adapter(
+        &self,
+        _loc: &DeploymentLocator,
+        _capabilities: &Self::NodeCapabilities,
+        _unified_api_version: UnifiedMappingApiVersion,
+        _stopwatch_metrics: StopwatchMetrics,
+    ) -> Result<Arc<Self::TriggersAdapter>, Error> {
+        let adapter = TriggersAdapter {};
+        Ok(Arc::new(adapter))
+    }
+
+    async fn new_firehose_block_stream(
+        &self,
+        deployment: DeploymentLocator,
+        store: Arc<dyn WritableStore>,
+        start_blocks: Vec<BlockNumber>,
+        filter: Arc<Self::TriggerFilter>,
+        metrics: Arc<BlockStreamMetrics>,
+        unified_api_version: UnifiedMappingApiVersion,
+    ) -> Result<Box<dyn BlockStream<Self>>, Error> {
+        let adapter = self
+            .triggers_adapter(
+                &deployment,
+                &NodeCapabilities {},
+                unified_api_version.clone(),
+                metrics.stopwatch.clone(),
+            )
+            .expect(&format!("no adapter for network {}", self.name,));
+
+        let firehose_endpoint = match self.firehose_endpoints.random() {
+            Some(e) => e.clone(),
+            None => return Err(anyhow::format_err!("no firehose endpoint available",)),
+        };
+
+        let logger = self
+            .logger_factory
+            .subgraph_logger(&deployment)
+            .new(o!("component" => "FirehoseBlockStream"));
+
+        let firehose_mapper = Arc::new(FirehoseMapper {});
+        let firehose_cursor = store.block_cursor();
+
+        Ok(Box::new(FirehoseBlockStream::new(
+            firehose_endpoint,
+            firehose_cursor,
+            firehose_mapper,
+            adapter,
+            filter,
+            start_blocks,
+            logger,
+        )))
+    }
+
+    async fn new_polling_block_stream(
+        &self,
+        _deployment: DeploymentLocator,
+        _writable: Arc<dyn WritableStore>,
+        _start_blocks: Vec<BlockNumber>,
+        _subgraph_start_block: Option<BlockPtr>,
+        _filter: Arc<Self::TriggerFilter>,
+        _metrics: Arc<BlockStreamMetrics>,
+        _unified_api_version: UnifiedMappingApiVersion,
+    ) -> Result<Box<dyn BlockStream<Self>>, Error> {
+        panic!("SOLANA does not support polling block stream")
```

Comment:
It was `NEAR` in capital before because it's own it's spelled, but for `SOLANA` should be `Solana`.